### PR TITLE
docs(Contributing): update squash guide to mention explicit branch

### PR DESCRIPTION
**Pull request by TheNando**
This isn't a GitHub Issues, but I think it's a good idea.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -191,7 +191,7 @@ Once we are happy with the final state of the PR, we will write "LGTM" or "Looks
   1. You may run into a merge conflict. In that case, open the conflicting file(s) and modify it so that it reflects the desired final state.
   2. `git add <filename>` will include the corrected file into the rebase
   3. `git rebase --continue` will conclude the rebase now that the conflict has been resolved
- 4. `git push -f` to force push your branch up to Github
+ 4. `git push -f origin <branch_name>` to force push your branch up to Github. Please don't push to master branch.
```
Comment (ericcarraway) [inline CONTRIBUTING.md]: +100 @TheNando  I lost some time on this today, as did @Droogans when he had to clean up after me.

**Comment by Droogans** [comment]
@TheNando maybe it's time we talk about using [squashed branches](http://andrew.yurisich.com/work/2015/07/25/squashed-branches-for-pull-requests/)?

I wrote the blog post linked here as a response to this PR since it's somewhat related. We've talked about this casually, but I'd like to switch to this. @parlarjb @megalord @CITguy do you think there are benefits to this, or is it too much work?

**Comment by parlarjb** [comment]
@Droogans The only issue I have with your squashed branches in a github context is that I personally find the PR discussions more useful than the commit messages.

If you want to use squashed branches, then I'd ask that the final PR contains a link to the original PR, so it's easy to find all the discussion that let us there.

File: CONTRIBUTING.md (hunk 1 above)
Comment (parlarjb) [inline CONTRIBUTING.md]: Ahh, I see the problem: When I push up my initial branch, I always do `git push -u origin`. This creates the remote branch, and sets the local one as a tracking branch of it. Thus, when I later do `git push -f`, git already knows to push to the remote branch name.

Comment (CITguy) [inline CONTRIBUTING.md]: I like where this is going, but where in the documentation is `origin` defined?  I'd suggest some documentation around setting up the `upstream` and `origin` remotes when cloning/forking the repo.

**Comment by parlarjb** [comment]
LGTM

**Pull request merged**